numpy/linalg/linalg.py

```diff
@@ -2141,7 +2141,10 @@ def lstsq(a, b, rcond="warn"):
 
     signature = 'DDd->Ddid' if isComplexType(t) else 'ddd->ddid'
     extobj = get_linalg_error_extobj(_raise_linalgerror_lstsq)
-    x, resids, rank, s = gufunc(a, b, rcond, signature=signature, extobj=extobj)
+    _b = zeros((m, n_rhs + 1), dtype=b.dtype) if is_empty_b else b
+    x, resids, rank, s = gufunc(a, _b, rcond, signature=signature, extobj=extobj)
+    x = x[:, :n_rhs]
+    resids = resids[:n_rhs] if resids.ndim == 1 else resids[:, :n_rhs]
```

Comment:
I think this would be clearer with an explicit branch on `is_empty_b` - since the slicing isn't needed normally anyway